samples/client-sdk/getting-started-daily-js/index.js

```diff
@@ -0,0 +1,389 @@
+/**
+ * Class managing Daily.co call functionalities including joining, leaving,
+ * handling audio/video tracks and UI updates for track states.
+ */
+class DailyCallManager {
+  constructor() {
+    this.call = DailyIframe.createCallObject();
+    this.isCameraMuted = null;
+    this.isMicMuted = null;
+    this.currentRoomUrl = null;
+    this.initialize();
+  }
+
+  /**
+   * Performs initial setup of event listeners and UI component interactions.
+   */
+  async initialize() {
+    await this.setupEventListeners();
+    document
+      .getElementById('toggle-camera')
+      .addEventListener('click', () => this.toggleCamera());
+    document
+      .getElementById('toggle-mic')
+      .addEventListener('click', () => this.toggleMicrophone());
+  }
+
+  /**
+   * Configures event listeners for various call-related events.
+   */
+  async setupEventListeners() {
+    const events = {
+      'active-speaker-change': this.handleActiveSpeakerChange.bind(this),
+      error: this.handleError.bind(this),
+      'joined-meeting': this.handleJoin.bind(this),
+      'left-meeting': this.handleLeave.bind(this),
+      'participant-joined': this.handleParticipantJoinLeft.bind(this),
+      'participant-left': this.handleParticipantJoinLeft.bind(this),
+      'participant-updated': this.handleParticipantStateUpdate.bind(this),
+      'track-started': this.displayTrack.bind(this),
+      'track-stopped': this.destroyTrack.bind(this),
+    };
+
+    // Register the event handlers with the call object.
+    Object.entries(events).forEach(([event, handler]) => {
+      this.call.on(event, handler);
+    });
+  }
+
+  /**
+   * Handler for the local participant joining:
+   * - Prints the room URL
+   * - Enables the toggle camera and mic buttons
+   * - Gets the initial track states
+   * - Sets up the device selectors
+   */
+  handleJoin() {
+    console.log(`Successfully joined: ${this.currentRoomUrl}`);
+
+    // Enable the toggle camera and mic buttons
+    document.getElementById('toggle-camera').disabled = false;
+    document.getElementById('toggle-mic').disabled = false;
+
+    // Set up the camera and mic selectors
+    this.setupDeviceSelectors();
+
+    // Get the initial track states
+    this.getTrackStates();
+
+    // Update the participant count
+    document.getElementById(
+      'participant-count'
+    ).textContent = `Participants: ${this.getParticipantCount()}`;
+  }
+
+  /**
+   * Handler for remote participants joining and leaving. Updates the participant count.
+   */
+  handleParticipantJoinLeft() {
+    document.getElementById(
```

Comment:
Optional nit: since you're duplicating this here and on line 70 might be nice to pull it out into a little method for reuse.